rfcs/0015_Arm_CMSIS-NN_Integration.md

```diff
@@ -0,0 +1,118 @@
+- Feature Name: [RFC] Use CMSIS-NN with TVM
+- Start Date: July 2021
+- RFC PR: https://github.com/apache/tvm-rfcs/pull/15
+- GitHub Issue: https://github.com/apache/tvm/issues/8646
+
+# Acronyms
+CMSIS: Common Microcontroller Software Interface Standard
+ACL: The Compute Library for the Arm® Architecture
+MLF: Model Library Format
+
+# Summary
+
+This RFC introduces plan of integration of CMSIS-NN library into TVM. It consists of efficient kernels targeted for Arm's Cortex-M architecture.
+
+Please refer to the following pages for more details on CMSIS-NN.
+https://arm-software.github.io/CMSIS_5/NN/html/index.html
+https://github.com/ARM-software/CMSIS_5/tree/develop/CMSIS/NN
+
+First PR in the series of PRs to fulfill this integration would be graph partitioner for softmax int8. Detailed plan can found below in this RFC.
+
+
+# Motivation
+
+CMSIS-NN library consists of hand-tuned kernels that are suitable for Cortex-M and are compliant with the quantization scheme used in Tensorflow Lite. They have been optimized for better performance and small memory footprint which is required on these embedded devices and it would make sense for TVM to reuse these while generating code for Cortex-M. They have been integrated with the TensorFlow Lite Micro project.
+
+
+# Guide-level explanation
```

Comment:
Better to include a little more details. For example,
- How to build with CMSIS-NN support (e.g., `config.cmake`).
- The user interface in Python in addition to TVMC.

Reply:
We are planning to implement it through the existing BYOC infrastructure. We will move to the target hooks when they become available. Running tests using EthosU test infrastructure is mentioned in the EthosU RFC already. We plan to reuse that for CMSIS-NN. If it takes a while in merging, then we can introduce unit tests for testing using tir. I've also added comments about config.cmake changes and python interfaces.

Reply:
I've made some suggestions to reorganize this section a bit to add details

Reply:
I've taken care of all the suggestions. Thanks for reviewing it multiple times Andrew. Really appreciate it.